README.md

```diff
@@ -138,14 +138,26 @@ Paths provided are relative to the current directory from which you execute the
 
 There are plenty of examples you can use to start off:
 
-- `example`: contains an example app with version 2 of the specification. It will give you an idea how to annotate your comments in order to include them in the output specification.
+- `examples/js`: contains an example app with version 2 of the specification. It will give you an idea how to annotate your comments in order to include them in the output specification.
+- `examples/ts`: contains an example TypeScript app with version 2 of the specification. 
 - `test/cli`: CLI tests you can read to get ideas about the available functionalities of the CLI. (apart from the obvious help menu)
 - `test/example`: various assets for tests you can also re-use for starting definitions, routes, etc.
 
 ### CLI
 
 On top of the Node API, there is also a [command line interface](./docs/CLI.md).
 
+
+### Using with TypeScript at Runtime
+
+Yes, you can use swagger-jsdoc with your TypeScript project. There are some adjustments in the configuration necessary, though.
+1. You can install types via `@types/swagger-jsdoc`.
+2. Typescript is usually transpiled to JavaScript. So in order to generate the Swagger document the `apis: [...]` option has to point to the transpiled JavaScript files, and not to the TypeScript source files. In default configuration, the transpiled files are located in the `dist` directory (and the files end with `.js`).
+3. Annotating the model classes with the swagger comments is a great idea. However, these comments have to survive the transpilation process. A short workaround is to add a dummy function that you annotate instead of the model classes. 
+
+An example canfiguration can be found in `examples/ts`.
```

Comment:
I'd suggest that we focus on clearing out the more important questions on the ways of integrating TS before moving to the folder organisation. I am not sure it's worth having a whole separate folder with examples in TS as the code base is written in vanilla nodejs and won't quite fit. We can shush the linter, of course, but examples in the codebase exist to be used in tests, not only referenced in the documentation.